code_examples/dapp/src/dapp/signPresentation.ts

```diff
@@ -1,38 +1,32 @@
-/* eslint-disable @typescript-eslint/no-unused-vars */
-import { blake2AsU8a, encodeAddress } from '@polkadot/util-crypto'
-import { Keyring } from '@polkadot/api'
-
 import * as Kilt from '@kiltprotocol/sdk-js'
 
-let claim: Kilt.IClaim
-let did: Kilt.DidDocument
-let keyring: Keyring
-
-export async function main() {
-  const credential = Kilt.Credential.fromClaim(claim)
-
-  const attestationKey = did.assertionMethod?.[0]
-  if (!attestationKey) {
-    return
-  }
+export type Parameter = {
+  didUri: Kilt.DidUri
+  assertionMethodKey: Kilt.KiltKeyringPair
+  domainLinkageCredential: Kilt.ICredential
+}
 
-  // Create a callback that uses the DID attestation key to sign the credential.
-  const signCallback: Kilt.SignCallback = async ({ data }) => {
-    const { publicKey, type } = attestationKey
-    const address = encodeAddress(
-      type === 'ecdsa' ? blake2AsU8a(publicKey) : publicKey,
-      Kilt.Utils.ss58Format
-    )
-    const keypair = keyring.getPair(address) as Kilt.KiltKeyringPair
-    return {
-      signature: keypair.sign(data),
-      keyType: keypair.type,
-      keyUri: `${did.uri}${attestationKey.id}`
-    }
+export async function main({
+  didUri,
+  assertionMethodKey,
+  domainLinkageCredential
+}: Parameter) {
+  // We need the KeyId of the AssertionMethod Key. There is only
+  // one AssertionMethodKey and its id is stored on the blockchain.
+  const didResolveResult = await Kilt.Did.resolve(didUri)
+  if (typeof didResolveResult.document === 'undefined') {
```

Comment:
```suggestion
  if (!didResolveResult.document) {
```

I think this is the way we check for undefined values.

Reply:
The reason for doing this `typeof` stuff is that this is the way to only check for undefined. The `!` also returns `true` in the following cases:

* ""
* 0
* null
* false
* NaN

And since undefined could be overwritten with something else, `typeof` is used instead of comparing.